chain/chain/src/block_processing_utils.rs

```diff
@@ -21,6 +23,12 @@ pub(crate) struct BlockPreprocessInfo {
     pub(crate) challenges_result: ChallengesResult,
     pub(crate) challenged_blocks: Vec<CryptoHash>,
     pub(crate) provenance: Provenance,
+    /// This field will be set when the apply_chunks has finished.
+    /// This is used to provide a way for caller to wait for the finishing of applying chunks of
+    /// a block
+    pub(crate) apply_chunks_done: Arc<OnceCell<()>>,
+    /// This is used to calculate block processing time metric
+    pub(crate) block_received_time: Instant,
```

Comment:
I'd call it 'block_start_processing_time' (as I've mentioned in another comment -- the receive suggests that it also includes 'waiting in the queue part'